Index: indicator-sound-0.7.7.0/src/Makefile.am
===================================================================
--- indicator-sound-0.7.7.0.orig/src/Makefile.am	2011-09-13 10:12:28.475604990 +0200
+++ indicator-sound-0.7.7.0/src/Makefile.am	2011-09-13 10:12:37.735650908 +0200
@@ -29,7 +29,7 @@
             dbus-shared-names.h
 
 libsoundmenu_la_CFLAGS = $(APPLET_CFLAGS) -Wall -Werror -DG_LOG_DOMAIN=\"Indicator-Sound\"
-libsoundmenu_la_LIBADD = $(APPLET_LIBS)
+libsoundmenu_la_LIBADD = $(APPLET_LIBS) -lm
 libsoundmenu_la_LDFLAGS = -module -avoid-version
 
 
