=== modified file 'indicator-sound/src/Makefile.am'
--- indicator-sound/src/Makefile.am	2012-02-14 18:41:28 +0000
+++ indicator-sound/src/Makefile.am	2012-02-14 18:41:43 +0000
@@ -33,7 +33,7 @@
 	$(COVERAGE_CFLAGS) \
 	-Wall -Werror -DG_LOG_DOMAIN=\"Indicator-Sound\"
 libsoundmenu_la_LIBADD = $(APPLET_LIBS) -lm
-libsoundmenu_la_LDFLAGS = 
+libsoundmenu_la_LDFLAGS = \  
 	$(COVERAGE_LDFLAGS) \
 	-module -avoid-version
 

