=== modified file 'configure.ac'
--- old/configure.ac	2012-01-13 13:43:00 +0000
+++ new/configure.ac	2012-02-01 03:15:46 +0000
@@ -42,7 +42,7 @@
 GTK_REQUIRED_VERSION=2.22
 GTK3_REQUIRED_VERSION=3.0
 INDICATOR_REQUIRED_VERSION=0.3.19
-DBUSMENUGTK_REQUIRED_VERSION=0.3.101
+DBUSMENUGTK_REQUIRED_VERSION=0.5.90
 POLKIT_REQUIRED_VERSION=0.92
 PULSE_AUDIO_REQUIRED_VERSION=0.9.19
 INDICATOR_DISPLAY_OBJECTS=0.1.11

=== modified file 'src/indicator-sound.c'
--- old/src/indicator-sound.c	2012-01-25 02:57:50 +0000
+++ new/src/indicator-sound.c	2012-02-01 03:20:44 +0000
@@ -23,11 +23,7 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <gdk/gdkkeysyms.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menu.h>
-#else
 #include <libdbusmenu-gtk/menu.h>
-#endif
 #include <libido/idoscalemenuitem.h>
 
 #include <gio/gio.h>

=== modified file 'src/metadata-widget.h'
--- old/src/metadata-widget.h	2011-06-22 19:19:49 +0000
+++ new/src/metadata-widget.h	2012-02-01 03:20:44 +0000
@@ -20,11 +20,7 @@
 #define __METADATA_WIDGET_H__
 
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 G_BEGIN_DECLS
 

=== modified file 'src/mute-widget.h'
--- old/src/mute-widget.h	2011-07-17 02:40:00 +0000
+++ new/src/mute-widget.h	2012-02-01 03:20:44 +0000
@@ -22,11 +22,7 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 #include <libindicator/indicator-object.h>
 
 G_BEGIN_DECLS

=== modified file 'src/transport-widget.h'
--- old/src/transport-widget.h	2011-06-22 19:19:49 +0000
+++ new/src/transport-widget.h	2012-02-01 03:20:44 +0000
@@ -20,11 +20,7 @@
 #define __TRANSPORT_WIDGET_H__
 
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 #include "common-defs.h"
 

=== modified file 'src/voip-input-widget.h'
--- old/src/voip-input-widget.h	2011-06-22 19:19:49 +0000
+++ new/src/voip-input-widget.h	2012-02-01 03:20:44 +0000
@@ -22,11 +22,7 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 
 G_BEGIN_DECLS
 

=== modified file 'src/volume-widget.h'
--- old/src/volume-widget.h	2011-06-22 19:19:49 +0000
+++ new/src/volume-widget.h	2012-02-01 03:20:44 +0000
@@ -22,11 +22,7 @@
 #include <glib.h>
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#if GTK_CHECK_VERSION(3, 0, 0)
-#include <libdbusmenu-gtk3/menuitem.h>
-#else
 #include <libdbusmenu-gtk/menuitem.h>
-#endif
 #include <libindicator/indicator-object.h>
 
 G_BEGIN_DECLS

