--- a/src/transport-widget.c	2011-09-09 12:57:51 +0000
+++ b/src/transport-widget.c	2011-11-16 18:34:46 +0000
@@ -561,15 +561,15 @@
 {
   TransportAction event = TRANSPORT_ACTION_NO_ACTION;
   
-  if (x > 67 && x < 112
+  if (x > 57 && x < 102
       && y > 12 && y < 40){
     event = TRANSPORT_ACTION_PREVIOUS;
   }
-  else if (x > 111 && x < 153
+  else if (x > 101 && x < 143
            && y > 5 && y < 47){
     event = TRANSPORT_ACTION_PLAY_PAUSE;
   }
-  else if (x > 152 && x < 197
+  else if (x > 142 && x < 187
            && y > 12 && y < 40){
     event = TRANSPORT_ACTION_NEXT;
   }   

