=== modified file 'src/sound-service-dbus.c'
--- src/sound-service-dbus.c	2011-04-01 12:59:32 +0000
+++ src/sound-service-dbus.c	2011-04-13 05:23:24 +0000
@@ -200,7 +200,8 @@
 {
   GError * error = NULL;
   if (!g_spawn_command_line_async("gnome-volume-control --page=applications", &error) &&
-      !g_spawn_command_line_async("xfce4-mixer", &error)) 
+      !g_spawn_command_line_async("gnome-control-center sound", &error) && 
+      !g_spawn_command_line_async("xfce4-mixer", &error))
   {
     g_warning("Unable to show dialog: %s", error->message);
     g_error_free(error);

