=== modified file 'src/mpris2-controller.vala'
Index: indicator-sound-0.8.4.1/src/mpris2-controller.vala
===================================================================
--- indicator-sound-0.8.4.1.orig/src/mpris2-controller.vala	2012-03-21 10:53:40.041652304 -0400
+++ indicator-sound-0.8.4.1/src/mpris2-controller.vala	2012-03-21 10:53:40.053652304 -0400
@@ -248,6 +248,9 @@
 
   private bool validate_playlists_details()
   {
+    if (this.playlists.ActivePlaylist == null){
+      return false;
+    }
     if (this.playlists.ActivePlaylist.valid == false){
       return false;
     }    
Index: indicator-sound-0.8.4.1/src/mpris2-controller.c
===================================================================
--- indicator-sound-0.8.4.1.orig/src/mpris2-controller.c	2012-03-21 10:53:40.037652304 -0400
+++ indicator-sound-0.8.4.1/src/mpris2-controller.c	2012-03-21 10:54:46.433654489 -0400
@@ -1244,73 +1244,88 @@
 	ActivePlaylistContainer* _tmp2_;
 	ActivePlaylistContainer* _tmp3_;
 	gboolean _tmp4_;
-	gboolean _tmp5_;
-	MprisPlaylists* _tmp6_;
+	MprisPlaylists* _tmp5_;
+	ActivePlaylistContainer* _tmp6_;
 	ActivePlaylistContainer* _tmp7_;
 	ActivePlaylistContainer* _tmp8_;
-	ActivePlaylistContainer* _tmp9_;
-	PlaylistDetails* _tmp10_;
-	gboolean _tmp11_;
-	gboolean _tmp12_ = FALSE;
-	MprisPlaylists* _tmp13_;
+	gboolean _tmp9_;
+	gboolean _tmp10_;
+	MprisPlaylists* _tmp11_;
+	ActivePlaylistContainer* _tmp12_;
+	ActivePlaylistContainer* _tmp13_;
 	ActivePlaylistContainer* _tmp14_;
-	ActivePlaylistContainer* _tmp15_;
-	ActivePlaylistContainer* _tmp16_;
-	PlaylistDetails* _tmp17_;
-	const char* _tmp18_;
-	gboolean _tmp19_;
-	gboolean _tmp26_;
+	PlaylistDetails* _tmp15_;
+	gboolean _tmp16_;
+	gboolean _tmp17_ = FALSE;
+	MprisPlaylists* _tmp18_;
+	ActivePlaylistContainer* _tmp19_;
+	ActivePlaylistContainer* _tmp20_;
+	ActivePlaylistContainer* _tmp21_;
+	PlaylistDetails* _tmp22_;
+	const char* _tmp23_;
+	gboolean _tmp24_;
+	gboolean _tmp31_;
 	g_return_val_if_fail (self != NULL, FALSE);
 	_tmp0_ = self->priv->_playlists;
 	_tmp1_ = mpris_playlists_get_ActivePlaylist (_tmp0_);
 	_tmp2_ = _tmp1_;
 	_tmp3_ = _tmp2_;
-	_tmp4_ = (*_tmp3_).valid;
-	_tmp5_ = _tmp4_ == FALSE;
+	_tmp4_ = _tmp3_ == NULL;
 	_active_playlist_container_free0 (_tmp3_);
-	if (_tmp5_) {
+	if (_tmp4_) {
 		result = FALSE;
 		return result;
 	}
-	_tmp6_ = self->priv->_playlists;
-	_tmp7_ = mpris_playlists_get_ActivePlaylist (_tmp6_);
+	_tmp5_ = self->priv->_playlists;
+	_tmp6_ = mpris_playlists_get_ActivePlaylist (_tmp5_);
+	_tmp7_ = _tmp6_;
 	_tmp8_ = _tmp7_;
-	_tmp9_ = _tmp8_;
-	_tmp10_ = (*_tmp9_).details;
-	_tmp11_ = _tmp10_ == NULL;
-	_active_playlist_container_free0 (_tmp9_);
-	if (_tmp11_) {
+	_tmp9_ = (*_tmp8_).valid;
+	_tmp10_ = _tmp9_ == FALSE;
+	_active_playlist_container_free0 (_tmp8_);
+	if (_tmp10_) {
+		result = FALSE;
+		return result;
+	}
+	_tmp11_ = self->priv->_playlists;
+	_tmp12_ = mpris_playlists_get_ActivePlaylist (_tmp11_);
+	_tmp13_ = _tmp12_;
+	_tmp14_ = _tmp13_;
+	_tmp15_ = (*_tmp14_).details;
+	_tmp16_ = _tmp15_ == NULL;
+	_active_playlist_container_free0 (_tmp14_);
+	if (_tmp16_) {
 		result = FALSE;
 		return result;
 	}
-	_tmp13_ = self->priv->_playlists;
-	_tmp14_ = mpris_playlists_get_ActivePlaylist (_tmp13_);
-	_tmp15_ = _tmp14_;
-	_tmp16_ = _tmp15_;
-	_tmp17_ = (*_tmp16_).details;
-	_tmp18_ = (*_tmp17_).path;
-	_tmp19_ = _tmp18_ == NULL;
-	_active_playlist_container_free0 (_tmp16_);
-	if (_tmp19_) {
-		_tmp12_ = TRUE;
+	_tmp18_ = self->priv->_playlists;
+	_tmp19_ = mpris_playlists_get_ActivePlaylist (_tmp18_);
+	_tmp20_ = _tmp19_;
+	_tmp21_ = _tmp20_;
+	_tmp22_ = (*_tmp21_).details;
+	_tmp23_ = (*_tmp22_).path;
+	_tmp24_ = _tmp23_ == NULL;
+	_active_playlist_container_free0 (_tmp21_);
+	if (_tmp24_) {
+		_tmp17_ = TRUE;
 	} else {
-		MprisPlaylists* _tmp20_;
-		ActivePlaylistContainer* _tmp21_;
-		ActivePlaylistContainer* _tmp22_;
-		ActivePlaylistContainer* _tmp23_;
-		PlaylistDetails* _tmp24_;
-		const gchar* _tmp25_;
-		_tmp20_ = self->priv->_playlists;
-		_tmp21_ = mpris_playlists_get_ActivePlaylist (_tmp20_);
-		_tmp22_ = _tmp21_;
-		_tmp23_ = _tmp22_;
-		_tmp24_ = (*_tmp23_).details;
-		_tmp25_ = (*_tmp24_).name;
-		_tmp12_ = _tmp25_ == NULL;
-		_active_playlist_container_free0 (_tmp23_);
+		MprisPlaylists* _tmp25_;
+		ActivePlaylistContainer* _tmp26_;
+		ActivePlaylistContainer* _tmp27_;
+		ActivePlaylistContainer* _tmp28_;
+		PlaylistDetails* _tmp29_;
+		const gchar* _tmp30_;
+		_tmp25_ = self->priv->_playlists;
+		_tmp26_ = mpris_playlists_get_ActivePlaylist (_tmp25_);
+		_tmp27_ = _tmp26_;
+		_tmp28_ = _tmp27_;
+		_tmp29_ = (*_tmp28_).details;
+		_tmp30_ = (*_tmp29_).name;
+		_tmp17_ = _tmp30_ == NULL;
+		_active_playlist_container_free0 (_tmp28_);
 	}
-	_tmp26_ = _tmp12_;
-	if (_tmp26_) {
+	_tmp31_ = _tmp17_;
+	if (_tmp31_) {
 		result = FALSE;
 		return result;
 	}
@@ -1381,7 +1396,7 @@
 		_tmp2_ = path;
 		_tmp3_ = e;
 		_tmp4_ = _tmp3_->message;
-		g_warning ("mpris2-controller.vala:280: Could not activate playlist %s because %s", (const gchar*) _tmp2_, _tmp4_);
+		g_warning ("mpris2-controller.vala:283: Could not activate playlist %s because %s", (const gchar*) _tmp2_, _tmp4_);
 		_g_error_free0 (e);
 	}
 	__finally13:
