=== modified file 'src/sound-service-dbus.c'
--- old/src/sound-service-dbus.c	2011-08-10 16:58:49 +0000
+++ new/src/sound-service-dbus.c	2012-02-09 21:32:58 +0000
@@ -228,7 +228,7 @@
 {
   GError * error = NULL;
   if (!g_spawn_command_line_async("gnome-volume-control --page=applications", &error) &&
-      !g_spawn_command_line_async("gnome-control-center sound", &error) && 
+      !g_spawn_command_line_async("gnome-control-center sound-nua", &error) && 
       !g_spawn_command_line_async("xfce4-mixer", &error))
   {
     g_warning("Unable to show dialog: %s", error->message);

