=== modified file 'src/sound-service-dbus.c'
--- old/src/sound-service-dbus.c	2012-03-16 17:59:34 +0000
+++ new/src/sound-service-dbus.c	2012-03-19 14:54:30 +0000
@@ -227,8 +227,14 @@
                             gpointer user_data)
 {
   GError * error = NULL;
+  gchar* cmd;
+  if (!g_strcmp0 (g_getenv ("XDG_CURRENT_DESKTOP"), "Unity"))
+    cmd = "gnome-control-center sound-nua";
+  else
+    cmd = "gnome-control-center sound";
+
   if (!g_spawn_command_line_async("gnome-volume-control --page=applications", &error) &&
-      !g_spawn_command_line_async("gnome-control-center sound", &error) && 
+          !g_spawn_command_line_async(cmd, &error) && 
       !g_spawn_command_line_async("xfce4-mixer", &error))
   {
     g_warning("Unable to show dialog: %s", error->message);

